### PR TITLE
bugfix, CausalConv1d, CausalConv2d

### PR DESCRIPTION
**Pull request by KarelVesely84**
- CausalConv1d : dilation=15 -> dilation=1 in forward()
- CausalConv2d :
   - the conv_padding can be applied separately to time axis and dim axis
   - in previous code, the output dim depends on `config.conv_padding`
   - in past we added 2 zero rows and 2 zero cols at the top-left corner -> now we add rows/cols from both sides (to have smaller artifacts)
   - the code should be compatible with older models (small shift of input of `CausalConv2d` should not be important)

### CODE REVIEW

**Comment by KarelVesely84** [comment]
please test it with some existing model before merging

File: src/models/extractors.py [hunk 1]
```diff
@@ -94,13 +94,24 @@ def __init__(self, config):
                 )
             ],
         )
+
+        # these values are backward compatible with `config.conv_padding = [1, 1]`
+        left_conv_padding = 2  # gets prepended to each line of feature matrix
```
Comment (Lakoc) [inline src/models/extractors.py]: this cannot be hardcoded here, must be propagated from config

File: src/models/streaming_modules.py [hunk 2]
```diff
@@ -50,7 +50,12 @@ def __init__(self, in_channels, out_channels, kernel_size, stride=1, padding=Non
         )
 
     def forward(self, inputs):
-        inputs = F.pad(inputs, (self.left_padding[1], 0, self.left_padding[0], 0))
+        padding_left = 2  # compatible with `conv_padding = [ 1, 1 ]` models,
```
Comment (Lakoc) [inline src/models/streaming_modules.py]: +1

**Review by Lakoc — CHANGES_REQUESTED** [review]
LGTM, please before merging update padding to be inferred from configs

File: src/models/extractors.py (hunk 1 above)
Comment (KarelVesely84) [reply src/models/extractors.py]: this convolution is not over time -> there could be 0 padding

and 2 is there fore backward compatibility with existing models -> do we need that ?

File: src/models/streaming_modules.py (hunk 2 above)
Comment (KarelVesely84) [reply src/models/streaming_modules.py]: dtto

**Comment by KarelVesely84** [comment]
updated, please review again

- non-causal models are backward compatible
- causal models are backward compatible

possible formats of same padding:
```
config.conv_padding = [ 1, 1 ]
config.conv_padding = [[ 1, 1 ], [ 1, 1 ]]  # `[[ time_pad, dim_pad ], ...]]`
```

**Comment by KarelVesely84** [comment]
will reopen against develop

**Pull request closed without merging**